cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreServiceUtils.java

```diff
@@ -191,26 +179,16 @@ public static HiveSupport checkHiveSupport(ClassLoader hiveClassLoader) {
         }
       }
 
-      // In Hive 12, CLIService.getOperationStatus returns OperationState.
-      // In Hive 13, CLIService.getOperationStatus returns OperationStatus.
-      Class cliServiceClass = usingCL.loadClass("org.apache.hive.service.cli.CLIService");
-      Class operationHandleCl = usingCL.loadClass("org.apache.hive.service.cli.OperationHandle");
-      @SuppressWarnings("unchecked")
-      Method getStatusMethod = cliServiceClass.getDeclaredMethod("getOperationStatus", operationHandleCl);
-
-      // Rowset is an interface in Hive 13, but a class in Hive 12
-      Class rowSetClass = usingCL.loadClass("org.apache.hive.service.cli.RowSet");
-
-      if (rowSetClass.isInterface()
-        && getStatusMethod.getReturnType() == usingCL.loadClass("org.apache.hive.service.cli.OperationStatus")) {
-        return HiveSupport.HIVE_13;
-      } else if (!rowSetClass.isInterface()
-        && getStatusMethod.getReturnType() == usingCL.loadClass("org.apache.hive.service.cli.OperationState")) {
+      String hiveVersion = HiveVersionInfo.getVersion();
+      if (hiveVersion.startsWith("0.12.")) {
         return HiveSupport.HIVE_12;
+      } else if (hiveVersion.startsWith("0.13.")) {
+        return HiveSupport.HIVE_13;
       }
+
       throw new RuntimeException("Hive distribution not supported. Set the configuration '" +
-                                 Constants.Explore.EXPLORE_ENABLED +
-                                 "' to false to start up without Explore.");
+                                   Constants.Explore.EXPLORE_ENABLED +
+                                   "' to false to start up without Explore.");
     } catch (RuntimeException e) {
```

Comment:
but catching a `RuntimeException` and throwing it is meaningless. Why catch and rethrow?